firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -522,13 +525,21 @@ static long parseTokenExpirationTimestamp(String expiresIn) {
         : Long.parseLong(expiresIn.substring(0, expiresIn.length() - 1));
   }
 
-  private static void logFisCommunicationError(HttpURLConnection conn) {
+  private static void logFisCommunicationError(HttpURLConnection conn, String firebaseOptions) {
     String logString = readErrorResponse(conn);
     if (!TextUtils.isEmpty(logString)) {
       Log.w(FIS_TAG, logString);
+      Log.w(FIS_TAG, firebaseOptions);
     }
   }
 
+  private String availableFirebaseOptions(
+      @Nullable String appId, @NonNull String apiKey, @NonNull String projectId) {
+    return String.format(
+        "Firebase options used while communicating with Firebase server APIs: %s, %s, %s",
```

Comment:
... %s, %s (%s)", apiKey, projectId, appId

Reply:
I would prefer this order, given that API key is the most important value and App ID is not always available.